app/Observers/TorrentObserver.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * NOTICE OF LICENSE.
+ *
+ * UNIT3D is open-sourced software licensed under the GNU Affero General Public License v3.0
+ * The details is bundled with this project in the file LICENSE.txt.
+ *
+ * @project    UNIT3D
+ *
+ * @license    https://www.gnu.org/licenses/agpl-3.0.en.html/ GNU Affero General Public License v3.0
+ * @author     HDVinnie
+ */
+
+namespace App\Observers;
+
+use App\Models\Torrent;
+use Illuminate\Support\Facades\Cache;
+
+class TorrentObserver
+{
+    /**
+     * Handle the Torrent "created" event.
+     *
+     * @param \App\Models\Torrent $torrent
+     *
+     * @return void
+     */
+    public function created(Torrent $torrent)
+    {
+        Cache::put("torrent.{$torrent->info_hash}", $torrent);
+    }
+
+    /**
+     * Handle the Torrent "retrieved" event.
+     *
+     * @param \App\Models\Torrent $torrent
+     *
+     * @return void
+     */
+    public function retrieved(Torrent $torrent)
+    {
+        Cache::add("torrent.{$torrent->info_hash}", $torrent);
```

Comment:
Why not use `put()` here, so that any time a record is fetched from the DB, the cache is updated with the latest data? The `add()` method won't update the cache if a record already exists at that key, in which case the cache would return stale data if the underlying DB record is updated.

Reply:
Good point.